app/frontends/capabilities.ts

```diff
@@ -0,0 +1,141 @@
+import HarmonyRequest from '../models/harmony-request';
+import { Response, NextFunction } from 'express';
+import { keysToLowerCase } from '../util/object';
+import { RequestValidationError } from '../util/errors';
+import { CmrCollection, getCollectionsByIds, getCollectionsByShortName, getVariablesForCollection } from '../util/cmr';
+import { addCollectionsToServicesByAssociation } from '../middleware/service-selection';
+import _ from 'lodash';
+import { ServiceConfig } from '../models/services/base-service';
+import { harmonyCollections } from '../models/services';
+
+interface CollectionCapabilities {
+  conceptId: string;
+  shortName: string;
+  variableSubset: boolean;
+  bboxSubset: boolean;
+  shapeSubset: boolean;
+  concatenate: boolean;
+  reproject: boolean;
+  outputFormats: string[];
+  services: ServiceConfig<unknown>[];
+  variables: string[];
+}
+
+/**
+ * Loads the collection info from CMR using short name or concept ID passed into the harmony
+ * request
+ *
+ * @param req - The request sent by the client
+ * @returns the collection info
+ * @throws RequestValidationError if no collection is found or parameters are invalid
+ */
+async function loadCollectionInfo(req: HarmonyRequest): Promise<CmrCollection> {
+  const query = keysToLowerCase(req.query);
+  const { collectionid, shortname } = query;
+  let collections;
+  let pickedCollection;
+  if (!collectionid && !shortname) {
+    throw new RequestValidationError('Must specify either collectionId or shortName');
+  } else if (collectionid && shortname) {
+    throw new RequestValidationError('Must specify only one of collectionId or shortName, not both');
+  } else if (collectionid) {
+    collections = await getCollectionsByIds([collectionid], req.accessToken);
+    if (collections.length === 0) {
+      const message = `${collectionid} must be a CMR collection identifier, but `
+      + 'we could not find a matching collection. Please make sure the collection ID'
+      + 'is correct and that you have access to it.';
+      throw new RequestValidationError(message);
```

Comment:
I made it consistent with the way we handle it everywhere else in the code when we do not find matching collections/variables/granules. Right now those are 400s instead of 404s. I can see arguments for both.

Reply:
I like 404 better too - went ahead and changed.